<!DOCTYPE html><!--  This site was created in Webflow. https://www.webflow.com  -->
<!--  Last Published: Wed Dec 20 2023 09:06:55 GMT+0000 (Coordinated Universal Time)  -->
<html data-wf-page="6564063d4e8bc25b760c410e" data-wf-site="6563ee05e531be60fdec3b67">
<head>
  <meta charset="utf-8">
  <title>Source Code</title>
  <meta content="Source Code" property="og:title">
  <meta content="Source Code" property="twitter:title">
  <meta content="width=device-width, initial-scale=1" name="viewport">
  <meta content="Webflow" name="generator">
  <link href="css/normalize.css" rel="stylesheet" type="text/css">
  <link href="css/webflow.css" rel="stylesheet" type="text/css">
  <link href="css/darrens-radical-site-6d2e47.webflow.css" rel="stylesheet" type="text/css">
  <link href="https://fonts.googleapis.com" rel="preconnect">
  <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin="anonymous">
  <script src="https://ajax.googleapis.com/ajax/libs/webfont/1.6.26/webfont.js" type="text/javascript"></script>
  <script type="text/javascript">WebFont.load({  google: {    families: ["Lato:100,100italic,300,300italic,400,400italic,700,700italic,900,900italic","Vollkorn:400,400italic,700,700italic","Akshar:regular"]  }});</script>
  <script type="text/javascript">!function(o,c){var n=c.documentElement,t=" w-mod-";n.className+=t+"js",("ontouchstart"in o||o.DocumentTouch&&c instanceof DocumentTouch)&&(n.className+=t+"touch")}(window,document);</script>
  <link href="images/favicon.ico" rel="shortcut icon" type="image/x-icon">
  <link href="images/webclip.png" rel="apple-touch-icon">
</head>
<body>
  <section>
    <div class="section-ultra-gradients">
      <div data-w-id="69443aec-5f95-4d5d-4676-1e7fbc73e301" class="ultra-gradient-wrapper-3">
        <div class="main-shapes-wrapper">
          <div style="-webkit-transform:translate3d(0vw, 0, 0) scale3d(1, 1, 1) rotateX(0) rotateY(0) rotateZ(-180deg) skew(0, 0);-moz-transform:translate3d(0vw, 0, 0) scale3d(1, 1, 1) rotateX(0) rotateY(0) rotateZ(-180deg) skew(0, 0);-ms-transform:translate3d(0vw, 0, 0) scale3d(1, 1, 1) rotateX(0) rotateY(0) rotateZ(-180deg) skew(0, 0);transform:translate3d(0vw, 0, 0) scale3d(1, 1, 1) rotateX(0) rotateY(0) rotateZ(-180deg) skew(0, 0)" class="shape-3"></div>
          <div class="shape-2"></div>
          <div class="shape-1"></div>
        </div>
        <div class="blending-group-wrapper">
          <div class="shape-8-2"></div>
          <div class="shape-9"></div>
          <div class="shape-4-3"></div>
          <div class="shape-5-3"></div>
          <div class="shape-7-2"></div>
          <div class="shape-6"></div>
        </div>
        <div class="background-noise-3">
          <div class="nav-menu">
            <div class="sidebar_content">
              <div class="sidebar_inner">
                <div class="sidebar_flex">
                  <div class="sidebar_links">
                    <a href="index.html" class="link">Home</a>
                    <a href="get-started.html" class="link">Get Started</a>
                    <a href="learn-more.html" class="link">Learn More</a>
                    <a href="source-code.html" aria-current="page" class="link w--current">Source Code</a>
                  </div>
                </div>
                <div class="sidebar_revealer"></div>
              </div>
            </div>
            <div data-w-id="3e1c95c7-af49-0c46-3b06-f0aea76e497d" class="burger_menu">
              <div data-is-ix2-target="1" class="lottie-animation" data-w-id="3e1c95c7-af49-0c46-3b06-f0aea76e497e" data-animation-type="lottie" data-src="https://uploads-ssl.webflow.com/65643f49a1c6e64258f6e3f1/65643f49a1c6e64258f6e434_lf30_editor_z2PYbc.json" data-loop="0" data-direction="1" data-autoplay="0" data-renderer="svg" data-default-duration="3.503503360802889" data-duration="0"></div>
            </div>
          </div>
        </div>
      </div>
      <div class="gradient-intro-wrapper">
        <div class="container-w1 gutter-outside">
          <div class="text-center">
            <h1 class="heading-9">Source Code</h1>
          </div>
        </div>
        <div class="container-w3 gutter-outside">
          <div class="text-center">
            <div class="intro-description">
              <p class="copy-medium">Explore the code and see the statistical functions and plots that were made using R!</p>
            </div>
          </div>
        </div>
        <a href="https://github.com/darrenzwc/protein-poker" data-w-id="30232825-5687-77b4-4f3a-4201879553c0" target="_blank" class="button-wrapper w-inline-block">
          <div class="button-text">Click Me!</div><img src="images/arrow-pointing-to-right.svg" alt="" style="-webkit-transform:translate3d(0, 0, 0) scale3d(0.8, 0.8, 1) rotateX(0) rotateY(0) rotateZ(0) skew(0, 0);-moz-transform:translate3d(0, 0, 0) scale3d(0.8, 0.8, 1) rotateX(0) rotateY(0) rotateZ(0) skew(0, 0);-ms-transform:translate3d(0, 0, 0) scale3d(0.8, 0.8, 1) rotateX(0) rotateY(0) rotateZ(0) skew(0, 0);transform:translate3d(0, 0, 0) scale3d(0.8, 0.8, 1) rotateX(0) rotateY(0) rotateZ(0) skew(0, 0);opacity:0" class="button-arrow">
        </a>
        <h3 class="heading-8"><strong>Circos Visualizer (Preview)</strong></h3>
        <pre class="code-block w-code-block" style="display:block;overflow-x:auto;background:#2b2b2b;color:#f8f8f2;padding:0.5em"><code class="language-r" style="white-space:pre"><span>library(shiny)
</span>library(ggvis)
library(dplyr)
<span>specimens &lt;- read.table(</span><span style="color:#abe338">&quot;data/specimen_paca_au.csv&quot;</span><span>, header = </span><span style="color:#f5ab35">TRUE</span><span>, sep = </span><span style="color:#abe338">&quot;\t&quot;</span><span>) %&gt;% filter(icgc_donor_id %in% use_donors)
</span><span>snp &lt;- read.table(</span><span style="color:#abe338">&quot;data/filt_simple_somatic_mutation_paca_au_2.csv&quot;</span><span>, header = </span><span style="color:#f5ab35">TRUE</span><span>, sep = </span><span style="color:#abe338">&quot;\t&quot;</span><span>) %&gt;% filter(icgc_donor_id %in% use_donors, chromosome != </span><span style="color:#abe338">&quot;MT&quot;</span><span>)
</span><span>struct &lt;- read.table(</span><span style="color:#abe338">&quot;data/filt_structural_somatic_mutation_paca_au.csv&quot;</span><span>, header = </span><span style="color:#f5ab35">TRUE</span><span>, sep = </span><span style="color:#abe338">&quot;\t&quot;</span><span>) %&gt;% filter(icgc_donor_id %in% use_donors)
</span><span>cnv &lt;- read.table(</span><span style="color:#abe338">&quot;data/filt_copy_number_somatic_mutation_paca_au.csv&quot;</span><span>, header = </span><span style="color:#f5ab35">TRUE</span><span>, sep = </span><span style="color:#abe338">&quot;\t&quot;</span><span>) %&gt;% filter(icgc_donor_id %in% use_donors)
</span><span>genes &lt;- read.csv(</span><span style="color:#abe338">&quot;data/genes.csv&quot;</span><span>)
</span><span></span><span style="color:#d4d0ab"># top_genes &lt;- xap.read_table(&quot;icgc_top_genes&quot;)</span><span>
</span><span>clinvar &lt;- read.csv(</span><span style="color:#abe338">&quot;data/filt_clinvar_variant_summary.csv&quot;</span><span>, header = </span><span style="color:#f5ab35">TRUE</span><span>, sep = </span><span style="color:#abe338">&quot;\t&quot;</span><span>)
</span>
snp_genes &lt;- snp %&gt;%
<span>  inner_join(genes, by = </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#abe338">&quot;gene_affected&quot;</span><span> = </span><span style="color:#abe338">&quot;ensembl_gene_id&quot;</span><span>)) %&gt;%
</span><span>  inner_join(clinvar, by = </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#abe338">&quot;hgnc_symbol&quot;</span><span> = </span><span style="color:#abe338">&quot;gene_symbol&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome&quot;</span><span> = </span><span style="color:#abe338">&quot;chromosome&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome_start&quot;</span><span> = </span><span style="color:#abe338">&quot;start&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome_end&quot;</span><span> = </span><span style="color:#abe338">&quot;stop&quot;</span><span>, </span><span style="color:#abe338">&quot;mutated_from_allele&quot;</span><span> = </span><span style="color:#abe338">&quot;reference_allele&quot;</span><span>, </span><span style="color:#abe338">&quot;mutated_to_allele&quot;</span><span> = </span><span style="color:#abe338">&quot;alternate_allele&quot;</span><span>))
</span>
<span>snp_transcripts &lt;- snp_genes[, </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#abe338">&quot;icgc_donor_id&quot;</span><span>, </span><span style="color:#abe338">&quot;hgnc_symbol&quot;</span><span>, </span><span style="color:#abe338">&quot;transcript_affected&quot;</span><span>, </span><span style="color:#abe338">&quot;gene_affected&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome_start&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome_end&quot;</span><span>, </span><span style="color:#abe338">&quot;mutated_from_allele&quot;</span><span>, </span><span style="color:#abe338">&quot;mutated_to_allele&quot;</span><span>, </span><span style="color:#abe338">&quot;consequence_type&quot;</span><span>, </span><span style="color:#abe338">&quot;mutation_type&quot;</span><span>)]
</span>
snp_transcripts &lt;- snp_transcripts[!duplicated(snp_transcripts), ]
<span>snp$is_clinvar &lt;- ifelse(</span><span style="color:#f5ab35">is.na</span><span>(snp_clinvar$hgnc_symbol), </span><span style="color:#abe338">&quot;No&quot;</span><span>, </span><span style="color:#abe338">&quot;Yes&quot;</span><span>)
</span>
<span></span><span style="color:#d4d0ab">## snp and struct are from the same specimens. cnv has more than one specimen per donor</span><span>
</span>specimens &lt;- unique(snp$icgc_specimen_id)
cnv &lt;- cnv %&gt;% filter(icgc_specimen_id %in% specimens)
<span>chroms &lt;- </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#f5ab35">1</span><span>:</span><span style="color:#f5ab35">22</span><span>, </span><span style="color:#abe338">&quot;X&quot;</span><span>, </span><span style="color:#abe338">&quot;Y&quot;</span><span>)
</span><span>lengths &lt;- </span><span style="color:#f5ab35">c</span><span>(
</span><span>  </span><span style="color:#f5ab35">249250621</span><span>, </span><span style="color:#f5ab35">243199373</span><span>, </span><span style="color:#f5ab35">198022430</span><span>, </span><span style="color:#f5ab35">191154276</span><span>, </span><span style="color:#f5ab35">180915260</span><span>, </span><span style="color:#f5ab35">171115067</span><span>,
</span><span>  </span><span style="color:#f5ab35">159138663</span><span>, </span><span style="color:#f5ab35">146364022</span><span>, </span><span style="color:#f5ab35">141213431</span><span>, </span><span style="color:#f5ab35">135534747</span><span>, </span><span style="color:#f5ab35">135006516</span><span>, </span><span style="color:#f5ab35">133851895</span><span>,
</span><span>  </span><span style="color:#f5ab35">115169878</span><span>, </span><span style="color:#f5ab35">107349540</span><span>, </span><span style="color:#f5ab35">102531392</span><span>, </span><span style="color:#f5ab35">90354753</span><span>, </span><span style="color:#f5ab35">81195210</span><span>, </span><span style="color:#f5ab35">78077248</span><span>,
</span><span>  </span><span style="color:#f5ab35">59128983</span><span>, </span><span style="color:#f5ab35">63025520</span><span>, </span><span style="color:#f5ab35">48129895</span><span>, </span><span style="color:#f5ab35">51304566</span><span>, </span><span style="color:#f5ab35">155270560</span><span>, </span><span style="color:#f5ab35">59373566</span><span>
</span>)
<span>radians_f &lt;- create_radians(chroms[</span><span style="color:#f5ab35">1</span><span>:</span><span style="color:#f5ab35">23</span><span>], lengths[</span><span style="color:#f5ab35">1</span><span>:</span><span style="color:#f5ab35">23</span><span>])
</span>radians_m &lt;- create_radians(chroms, lengths)
<span>server &lt;- </span><span style="color:#dcc6e0">function</span><span>(input, output, session) {
</span>  re_values &lt;- reactiveValues(
<span>    scaling_factors = </span><span style="color:#f5ab35">rep</span><span>(</span><span style="color:#f5ab35">1</span><span>, </span><span style="color:#f5ab35">24</span><span>), previous_radians = radians_f,
</span><span>    chrom_clicked = </span><span style="color:#abe338">&quot;1&quot;</span><span>, chroms_selected = </span><span style="color:#f5ab35">NULL</span><span>
</span>  )
  radians &lt;- reactive({
    donors_filt &lt;- donors %&gt;% filter(icgc_donor_id == input$donor)
    gender &lt;- unique(donors_filt$donor_sex)
<span>    isolate(mid &lt;- mean(rads[</span><span style="color:#f5ab35">names</span><span>(rads) == re_values$chrom_clicked]))
</span>
<span>    isolate(prev_mid &lt;- mean(re_values$previous_radians[</span><span style="color:#f5ab35">names</span><span>(rads) == re_values$chrom_clicked]))
</span>
    offset &lt;- mid - prev_mid
    rads - offset
  })
  track_radians &lt;- reactive({
<span>    create_track_radians(radians(), points_per_track = </span><span style="color:#f5ab35">rep</span><span>(</span><span style="color:#f5ab35">20</span><span>, </span><span style="color:#f5ab35">length</span><span>(radians())))
</span>  })
  output$top_clinvar_genes &lt;- renderTable(
    {
<span>      top_genes &lt;- top_genes[, </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#abe338">&quot;hgnc_symbol&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome_start&quot;</span><span>, </span><span style="color:#abe338">&quot;mutated_from_allele&quot;</span><span>, </span><span style="color:#abe338">&quot;mutated_to_allele&quot;</span><span>, </span><span style="color:#abe338">&quot;consequence_type&quot;</span><span>, </span><span style="color:#abe338">&quot;count&quot;</span><span>)] %&gt;%
</span><span>        mutate(count = </span><span style="color:#f5ab35">as.integer</span><span>(count), chromosome_start = </span><span style="color:#f5ab35">as.integer</span><span>(chromosome_start)) %&gt;%
</span>        arrange(desc(count)) %&gt;%
<span>        top_n(n = </span><span style="color:#f5ab35">10</span><span>)
</span><span>      setNames(top_genes, </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#abe338">&quot;HGNC&quot;</span><span>, </span><span style="color:#abe338">&quot;Chr&quot;</span><span>, </span><span style="color:#abe338">&quot;Start&quot;</span><span>, </span><span style="color:#abe338">&quot;From&quot;</span><span>, </span><span style="color:#abe338">&quot;To&quot;</span><span>, </span><span style="color:#abe338">&quot;Consequence&quot;</span><span>, </span><span style="color:#abe338">&quot;Count&quot;</span><span>))
</span>    },
<span>    include.rownames = </span><span style="color:#f5ab35">FALSE</span><span>
</span>  )
  seq_df &lt;- reactive({
    donors_filt &lt;- donors %&gt;% filter(icgc_donor_id == input$donor)
<span>    </span><span style="color:#d4d0ab">#    gender &lt;- unique(donors_filt$donor_sex)</span><span>
</span>
<span>    </span><span style="color:#d4d0ab">#     if (gender == &quot;female&quot;) {</span><span>
</span><span>    </span><span style="color:#d4d0ab">#       scale &lt;- re_values$scaling_factors[1:23]</span><span>
</span><span>    </span><span style="color:#d4d0ab">#     } else {</span><span>
</span>    scale &lt;- re_values$scaling_factors
<span>    </span><span style="color:#d4d0ab">#    }</span><span>
</span>
    create_seq_df(radians(), scale = scale)
  })
  snp_plot_data &lt;- reactive({
    snp_filt &lt;- snp %&gt;% filter(icgc_donor_id == input$donor, consequence_type %in% input$consequence_type)
    snp_data &lt;- snp_filt %&gt;%
      group_by(
        chromosome, chromosome_start, chromosome_end, gene_affected,
        mutation_type, mutated_from_allele, mutated_to_allele, is_clinvar
      ) %&gt;%
      summarise(transcripts = n())
    points &lt;- fit_points(snp_data$chromosome, snp_data$chromosome_start, snp_data$transcripts,
<span>      </span><span style="color:#f5ab35">0.8</span><span>, </span><span style="color:#f5ab35">0.6</span><span>, seq_df(),
</span><span>      metadata = snp_data[, </span><span style="color:#f5ab35">c</span><span>(
</span><span>        </span><span style="color:#abe338">&quot;transcripts&quot;</span><span>, </span><span style="color:#abe338">&quot;mutation_type&quot;</span><span>, </span><span style="color:#abe338">&quot;gene_affected&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;chromosome_start&quot;</span><span>, </span><span style="color:#abe338">&quot;chromosome_end&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;mutated_from_allele&quot;</span><span>, </span><span style="color:#abe338">&quot;mutated_to_allele&quot;</span><span>, </span><span style="color:#abe338">&quot;is_clinvar&quot;</span><span>
</span>      )],
<span>      min_value = </span><span style="color:#f5ab35">0</span><span>, max_value = </span><span style="color:#f5ab35">max</span><span>(snp_data$transcripts) + </span><span style="color:#f5ab35">1</span><span>
</span>    )
<span>    points$id &lt;- paste0(</span><span style="color:#abe338">&quot;snp&quot;</span><span>, </span><span style="color:#f5ab35">1</span><span>:nrow(points))
</span>
    points
  })
  clinvar_data &lt;- reactive({
    clinvar_gene_data &lt;- snp_genes %&gt;% filter(icgc_donor_id == input$donor)
    clinvar_gene_data
  })
  snp_transcript_data &lt;- reactive({
    snp_data &lt;- snp %&gt;% filter(icgc_donor_id == input$donor)
    snp_data
  })
  struct_plot_data &lt;- reactive({
    struct_filt &lt;- struct %&gt;% filter(
      icgc_donor_id == input$donor,
<span>      chr_from != chr_to | </span><span style="color:#f5ab35">abs</span><span>(chr_from_bkpt - chr_to_bkpt) &gt; </span><span style="color:#f5ab35">10</span><span>^</span><span style="color:#f5ab35">6</span><span>
</span>    )
    links &lt;- fit_links(
      struct_filt$chr_from, struct_filt$chr_to, struct_filt$chr_from_bkpt, struct_filt$chr_to_bkpt,
<span>      seq_df(), </span><span style="color:#f5ab35">0.6</span><span>, </span><span style="color:#f5ab35">0.6</span><span>, </span><span style="color:#f5ab35">0</span><span>
</span>    )
    links &lt;- links %&gt;%
      ungroup() %&gt;%
<span>      mutate(link = </span><span style="color:#f5ab35">as.numeric</span><span>(link)) %&gt;%
</span>      arrange(link) %&gt;%
      group_by(link)
<span>    links$annotation &lt;- </span><span style="color:#f5ab35">rep</span><span>(struct_filt$annotation, each = </span><span style="color:#f5ab35">3</span><span>)
</span><span>    links$pos_from &lt;- </span><span style="color:#f5ab35">rep</span><span>(struct_filt$chr_from_bkpt, each = </span><span style="color:#f5ab35">3</span><span>)
</span><span>    links$pos_to &lt;- </span><span style="color:#f5ab35">rep</span><span>(struct_filt$chr_to_bkpt, each = </span><span style="color:#f5ab35">3</span><span>)
</span>  })
  text_df &lt;- reactive({
<span>    seq_df() %&gt;% mutate(theta = (seq_start + seq_end) / </span><span style="color:#f5ab35">2</span><span>, r = </span><span style="color:#f5ab35">1.05</span><span>)
</span>  })
<span>  tooltip_fun &lt;- </span><span style="color:#dcc6e0">function</span><span>(data, session) {
</span><span>    </span><span style="color:#dcc6e0">if</span><span> (</span><span style="color:#abe338">&quot;mutation_type&quot;</span><span> %in% </span><span style="color:#f5ab35">names</span><span>(data)) {
</span>      tt_data &lt;- snp_plot_data()
      row &lt;- tt_data[tt_data$id == data$id, ]
      transcripts &lt;- snp_transcript_data() %&gt;% filter(gene_affected == row$gene_affected)
      gene_name &lt;- genes %&gt;% filter(ensembl_gene_id %in% row$gene_affected)
      clin_data &lt;- clinvar_data() %&gt;% filter(gene_affected %in% row$gene_affected)
        }
<span>      } </span><span style="color:#dcc6e0">else</span><span> {
</span><span>        clin_text &lt;- paste0(row$gene_affected, </span><span style="color:#abe338">&quot; | chr&quot;</span><span>, row$chromosome, </span><span style="color:#abe338">&quot;: No associated ClinVar records found.&quot;</span><span>)
</span>      }
<span>      transcript_text &lt;- </span><span style="color:#abe338">&quot;&quot;</span><span>
</span>
<span>      ensembl_gene_url &lt;- </span><span style="color:#abe338">&quot;http://www.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=&quot;</span><span>
</span><span>      ensembl_transcript_url &lt;- </span><span style="color:#abe338">&quot;http://www.ensembl.org/Homo_sapiens/Transcript/Summary?db=core;t=&quot;</span><span>
</span>
<span>      </span><span style="color:#dcc6e0">for</span><span> (i </span><span style="color:#dcc6e0">in</span><span> </span><span style="color:#f5ab35">1</span><span>:nrow(transcripts)) {
</span><span>        transcript_text &lt;- paste0(transcript_text, </span><span style="color:#abe338">&quot; &lt;a href=&quot;</span><span>, ensembl_transcript_url, transcripts[i, </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#abe338">&quot;transcript_affected&quot;</span><span>)], </span><span style="color:#abe338">&quot; target=_blank&gt;&quot;</span><span>, transcripts[i, </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#abe338">&quot;transcript_affected&quot;</span><span>)], </span><span style="color:#abe338">&quot;&lt;/a&gt;&quot;</span><span>)
</span>      }
<span>      record_colour &lt;- paste0(</span><span style="color:#abe338">&quot;rgb(&quot;</span><span>, sample(</span><span style="color:#f5ab35">170</span><span>:</span><span style="color:#f5ab35">255</span><span>, </span><span style="color:#f5ab35">1</span><span>), </span><span style="color:#abe338">&quot;,&quot;</span><span>, sample(</span><span style="color:#f5ab35">150</span><span>:</span><span style="color:#f5ab35">255</span><span>, </span><span style="color:#f5ab35">1</span><span>), </span><span style="color:#abe338">&quot;,&quot;</span><span>, sample(</span><span style="color:#f5ab35">170</span><span>:</span><span style="color:#f5ab35">255</span><span>, </span><span style="color:#f5ab35">1</span><span>), </span><span style="color:#abe338">&quot;)&quot;</span><span>)
</span>
      paste0(
<span>        </span><span style="color:#abe338">&quot;Start: &quot;</span><span>, row$chromosome_start, </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;End: &quot;</span><span>, row$chromosome_end, </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;Base Change: &quot;</span><span>, row$mutated_from_allele, </span><span style="color:#abe338">&quot;&gt;&quot;</span><span>, row$mutated_to_allele, </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;Mutation Type: &quot;</span><span>, row$mutation_type, </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;Gene Affected: &quot;</span><span>, paste0(</span><span style="color:#abe338">&quot;&lt;a href=&quot;</span><span>, ensembl_gene_url, row$gene_affected, </span><span style="color:#abe338">&quot; target=_blank&gt;&quot;</span><span>, 
</span><span>    } </span><span style="color:#dcc6e0">else</span><span> </span><span style="color:#dcc6e0">if</span><span> (</span><span style="color:#abe338">&quot;copy_number&quot;</span><span> %in% </span><span style="color:#f5ab35">names</span><span>(data)) {
</span>      tt_data &lt;- cnv_plot_data()
<span>      rows &lt;- tt_data[tt_data$theta &gt; data$theta - </span><span style="color:#f5ab35">0.0001</span><span> &amp; tt_data$theta &lt; data$theta + </span><span style="color:#f5ab35">0.0001</span><span>, ]
</span>
      paste0(
<span>        </span><span style="color:#abe338">&quot;Start: &quot;</span><span>, unique(rows$chromosome_start), </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;End: &quot;</span><span>, unique(rows$chromosome_end), </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;Segment Median &quot;</span><span>, unique(rows$segment_median), </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;Copy Number: &quot;</span><span>, sub(</span><span style="color:#abe338">&quot;cnv&quot;</span><span>, </span><span style="color:#abe338">&quot;&quot;</span><span>, unique(rows$copy_number)), </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;Mutation Type: &quot;</span><span>, unique(rows$mutation_type)
</span>      )
<span>    } </span><span style="color:#dcc6e0">else</span><span> </span><span style="color:#dcc6e0">if</span><span> (</span><span style="color:#abe338">&quot;link&quot;</span><span> %in% </span><span style="color:#f5ab35">names</span><span>(data)) {
</span>      tt_data &lt;- struct_plot_data()
<span>      row &lt;- tt_data[tt_data$link == data$link, ][</span><span style="color:#f5ab35">1</span><span>, ]
</span>
      paste0(
<span>        </span><span style="color:#abe338">&quot;&lt;center&gt;&quot;</span><span>,
</span><span>        row$name_from, </span><span style="color:#abe338">&quot;:&quot;</span><span>, row$pos_from, </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;&amp;rarr;&quot;</span><span>, </span><span style="color:#abe338">&quot;&lt;br&gt;&quot;</span><span>,
</span><span>        row$name_to, </span><span style="color:#abe338">&quot;:&quot;</span><span>, row$pos_to,
</span><span>        </span><span style="color:#abe338">&quot;&lt;center/&gt;&quot;</span><span>,
</span><span>        </span><span style="color:#abe338">&quot;Annotation: &quot;</span><span>, row$annotation
</span>      )
    }
  }
<span>  tooltip_click_fun &lt;- </span><span style="color:#dcc6e0">function</span><span>(data) {
</span>    str(data)
  }
    isolate(re_values$chrom_clicked &lt;- data$group)
    isolate(re_values$previous_radians &lt;- radians())
<span>    isolate(re_values$chroms_selected &lt;- chroms[which(re_values$scaling_factors &gt; </span><span style="color:#f5ab35">1</span><span>)])
</span>
    print(data)
  }
<span>  stroke_domain &lt;- </span><span style="color:#f5ab35">c</span><span>(
</span><span>    </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#f5ab35">1</span><span>:</span><span style="color:#f5ab35">22</span><span>, </span><span style="color:#abe338">&quot;X&quot;</span><span>, </span><span style="color:#abe338">&quot;Y&quot;</span><span>), </span><span style="color:#d4d0ab"># chromosomes</span><span>
</span><span>    paste0(</span><span style="color:#abe338">&quot;cnv&quot;</span><span>, </span><span style="color:#f5ab35">0</span><span>:</span><span style="color:#f5ab35">8</span><span>), </span><span style="color:#d4d0ab"># copy numbers</span><span>
</span><span>    </span><span style="color:#abe338">&quot;Yes&quot;</span><span>, </span><span style="color:#abe338">&quot;No&quot;</span><span>
</span><span>  ) </span><span style="color:#d4d0ab"># interchromosomal or not</span><span>
</span>
<span>  stroke_range &lt;- </span><span style="color:#f5ab35">c</span><span>(
</span><span>    </span><span style="color:#d4d0ab"># chromosome colours from Circos</span><span>
</span><span>    </span><span style="color:#abe338">&quot;#996600&quot;</span><span>, </span><span style="color:#abe338">&quot;#666600&quot;</span><span>, </span><span style="color:#abe338">&quot;#99991E&quot;</span><span>, </span><span style="color:#abe338">&quot;#CC0000&quot;</span><span>, </span><span style="color:#abe338">&quot;#FF0000&quot;</span><span>, </span><span style="color:#abe338">&quot;#FF00CC&quot;</span><span>, </span><span style="color:#abe338">&quot;#FFCCCC&quot;</span><span>, </span><span style="color:#abe338">&quot;#FF9900&quot;</span><span>, </span><span style="color:#abe338">&quot;#FFCC00&quot;</span><span>,
</span><span>    </span><span style="color:#abe338">&quot;#FFFF00&quot;</span><span>, </span><span style="color:#abe338">&quot;#CCFF00&quot;</span><span>, </span><span style="color:#abe338">&quot;#00FF00&quot;</span><span>, </span><span style="color:#abe338">&quot;#358000&quot;</span><span>, </span><span style="color:#abe338">&quot;#0000CC&quot;</span><span>, </span><span style="color:#abe338">&quot;#6699FF&quot;</span><span>, </span><span style="color:#abe338">&quot;#99CCFF&quot;</span><span>, </span><span style="color:#abe338">&quot;#00FFFF&quot;</span><span>, </span><span style="color:#abe338">&quot;#CCFFFF&quot;</span><span>,
</span><span>    </span><span style="color:#abe338">&quot;#9900CC&quot;</span><span>, </span><span style="color:#abe338">&quot;#CC33FF&quot;</span><span>, </span><span style="color:#abe338">&quot;#CC99FF&quot;</span><span>, </span><span style="color:#abe338">&quot;#666666&quot;</span><span>, </span><span style="color:#abe338">&quot;#999999&quot;</span><span>, </span><span style="color:#abe338">&quot;#CCCCCC&quot;</span><span>,
</span>
<span>    </span><span style="color:#d4d0ab"># colours for copy numbers</span><span>
</span><span>    </span><span style="color:#abe338">&quot;#0C2F60&quot;</span><span>, </span><span style="color:#abe338">&quot;#0C2F60&quot;</span><span>, </span><span style="color:#abe338">&quot;green&quot;</span><span>, </span><span style="color:#f5ab35">rep</span><span>(</span><span style="color:#abe338">&quot;#C11F29&quot;</span><span>, </span><span style="color:#f5ab35">6</span><span>),
</span>
<span>    </span><span style="color:#d4d0ab"># colours for links</span><span>
</span><span>    </span><span style="color:#abe338">&quot;blue&quot;</span><span>, </span><span style="color:#abe338">&quot;grey&quot;</span><span>
</span>  )
<span>  add_tooltip &lt;- </span><span style="color:#dcc6e0">function</span><span>(vis, html, on = </span><span style="color:#f5ab35">c</span><span>(</span><span style="color:#abe338">&quot;hover&quot;</span><span>, </span><span style="color:#abe338">&quot;click&quot;</span><span>)) {
</span>    on &lt;- match.arg(on)</code></pre>
      </div>
    </div>
  </section>
  <section class="footer-dark">
    <div class="container-2">
      <div class="footer-wrapper">
        <div class="footer-content">
          <div id="w-node-_6bf8dcfb-3925-e48a-7c84-eee50f60990c-0f609906" class="footer-block">
            <div class="title-small">Protein Poker</div>
            <a href="index.html" class="footer-link">Home</a>
            <a href="get-started.html" class="footer-link">Get Started</a>
            <a href="learn-more.html" class="footer-link">Curiosity</a>
            <a href="source-code.html" aria-current="page" class="footer-link w--current">Source-Code</a>
          </div>
          <div id="w-node-_6bf8dcfb-3925-e48a-7c84-eee50f609915-0f609906" class="footer-block">
            <div class="title-small">Mutate</div>
            <a href="https://darrenzwc.shinyapps.io/Sublime/" class="footer-link">6J5T</a>
            <a href="https://darrenzwc.shinyapps.io/5ieb/" class="footer-link">5IEB</a>
            <a href="https://darrenzwc.shinyapps.io/1kxp/" class="footer-link">1KXP</a>
            <a href="https://darrenzwc.shinyapps.io/vh4/" class="footer-link">1VH4</a>
          </div>
        </div>
      </div>
    </div>
    <div class="footer-copyright-center">Protein Poker © 2023</div>
  </section>
  <script src="https://d3e54v103j8qbb.cloudfront.net/js/jquery-3.5.1.min.dc5e7f18c8.js?site=6563ee05e531be60fdec3b67" type="text/javascript" integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin="anonymous"></script>
  <script src="js/webflow.js" type="text/javascript"></script>
</body>
</html>